File: DesafioConcreteSolutions/app/src/main/java/br/com/desafio/domain/Repo.java
package br.com.desafio.domain;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PUBLIC, suppressConstructorProperties = true)
public class Repo implements Serializable {
    private String name;
    @SerializedName("full_name")
    private String fullName;
}

File: domain/src/main/java/com/diegoferreiracaetano/domain/repo/Repo.kt
package com.diegoferreiracaetano.domain.repo

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.diegoferreiracaetano.domain.owner.Owner
import com.google.gson.annotations.SerializedName
import java.io.Serializable

@Entity(tableName = "repo",
        indices = arrayOf(Index(value = ["starts"])))
data class Repo (
        @PrimaryKey
        var id:Long,
        var name:String,
        var description:String?,
        @SerializedName("stargazers_count")
        var starts:Int,
        @SerializedName("forks_count")
        var forks:Int,
        @Embedded(prefix = "owner_")
        var owner: Owner) :Serializable{
        constructor():this(0,"","",0,0,Owner())

}
